src/Model/List/SS_List.php

```diff
@@ -15,83 +16,243 @@
+     *          // SQL: WHERE ("Name" = 'bob' OR "Age" = '21')
+     * @example // bob or anyone aged 21 or 43 in the list
+     *          $list = $list->filterAny(array('Name'=>'bob, 'Age'=>array(21, 43)));
+     *          // SQL: WHERE ("Name" = 'bob' OR ("Age" IN ('21', '43'))
+     * @example // all bobs, phils or anyone aged 21 or 43 in the list
+     *          $list = $list->filterAny(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+     *          // SQL: WHERE (("Name" IN ('bob', 'phil')) OR ("Age" IN ('21', '43'))
+     *
+     * @param string|array See {@link filter()}
+     * @return SS_List<T>
+     */
+    public function filterAny(...$args): SS_List;
+
+    /**
+     * Return a new instance of this list that excludes any items with these characteristics
+     *
+     * @example $list = $list->exclude('Name', 'bob'); // exclude bob from list
+     * @example $list = $list->exclude('Name', array('aziz', 'bob'); // exclude aziz and bob from list
+     * @example $list = $list->exclude(array('Name'=>'bob, 'Age'=>21)); // exclude bob that has Age 21
+     * @example $list = $list->exclude(array('Name'=>'bob, 'Age'=>array(21, 43))); // exclude bob with Age 21 or 43
+     * @example $list = $list->exclude(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+     *          // bob age 21 or 43, phil age 21 or 43 would be excluded
+     *
+     * @return SS_List<T>
+     */
+    public function exclude(...$args): SS_List;
+
+    /**
+     * Return a copy of this list which does not contain any items with any of these params
+     *
+     * @return SS_List<T>
+     */
+    public function excludeAny(...$args): SS_List;
+
+    /**
+     * Return a new instance of this list that excludes any items with these characteristics
+     * Filter this List by a callback function. The function will be passed each record of the List in turn,
+     * and must return true for the record to be included. Returns the filtered list.
+     *
+     * @example $list = $list->filterByCallback(function($item, $list) { return $item->Age == 9; })
+     * @return SS_List<T>
+     */
+    public function filterByCallback(callable $callback): SS_List;
+
+    /**
+     * Return the first item with the given ID
+     *
+     * @param int $id
+     * @return T|null
+     */
+    public function byID(int|string|null $id): mixed;
+    // Note that string ID's and null values should be handled as things like
+    // form submissions and controller params will often be string IDs,
+    // and null is often unintentionally passed in a lot of instances.
+    // If this is changed in the future to only accept int, ensure that all other
+    // filtering methods are updated at the same time for consistency
+
+    /**
+     * Filter this list to only contain the given Primary IDs
+     *
+     * @param array $ids Array of integers
+     * @return SS_List<T>
+     */
+    public function byIDs(array $ids): SS_List;
```

Comment:
I don't think these methods belong on a generic list interface.

Reply:
They're from Filterable, purpose of the card was to merge all the of these interfaces together

Reply:
I think we can allow ourselves to exclude methods that don't make sense in a generic list interface. They didn't make sense in a generic filterable interface either.

Excluding them also allows us to include appropriate typings for the various classes that implement this interface, as pointed out in various comments in this PR.

That said, to avoid holding onto this PR forever, I'll not hold up merging if you decide for some reason these methods _are_ appropriate for a generic list interface and choose to keep them.

Reply:
> However even that still has this:

That's because a) our predecessors haven't always treated it as agnostically as they perhaps should have, and added some DataObject specific methods to it, and b) it implemented `Filterable` and therefore _had_ to have some implementation for that method.

In any case, as I said, to avoid holding onto this PR forever, I'll not hold up merging if you decide for some reason these methods are appropriate for a generic list interface and choose to keep them.